### ISSUE TITLE
Weird bug

### ISSUE THREAD

**Issue by Worldgate** [cadon/ARKStatsExtractor#1365]
I'm using ASB Export Gun on ASE, fully updated, and for some reason after the 12'th import i get that error.  It doesn't give me the error if i do export dino by hand, but if i use the gun, that error.

![image](https://github.com/user-attachments/assets/1450d3d6-869f-41ef-8480-a331e213af0e)

**Comment by cadon** [comment]
- Is the file imported correctly if you drag&drop it onto ASB?
- Does it always happen after the 12th Gun-import?

It could be the file is not yet written completely yet so ASB can only read a part of that file.

**Comment by Worldgate** [comment]
the file is written differently though, it could be the mod, i'll get some example files.  Sadly if it is the mod, i doubt the mod author will even bother to update it since it's ASE and the modding community abandoned us a long time ago.

**Comment by Worldgate** [comment]
[Giganotosaurus_365176379-340077477.zip](https://github.com/user-attachments/files/17923349/Giganotosaurus_365176379-340077477.zip)

**Comment by Worldgate** [comment]
Giganotosaurus_365176379-340077477.sav is written correctly, but the second file seems weird in comparison.

**Comment by cadon** [comment]
I think I fixed it. The error was caused by a different encoding (ASB expects the string in utf8, but the failing file contained a string encoded in utf16, due to chinese characters in an ancestor's name).
The encoded json string has 4 leading bytes which contain the length of the string. Apparently when encoded in utf16, that length is the negative length of utf16 chars, probably to distinguish the encodings. At least that's what I hope and it handles the file you sent correctly. I'll release the fix in the next couple of days.
